### PR TITLE
Compatibility with Spring Boot 3.x

### PR DESCRIPTION
**Pull request by gedeffe**
In order to use gitlab4j-api with new version of SpringBoot and related dependencies, we had to upgrade several libraries.
It implies also a move from javax.xxx to jakarta.xxx for JEE libraries.

Relates to existing issues:
- Move to Java 11 or Java 17 for next major iteration of gitlab4j-api #817
- Add java9+ and Jakarta EE Support #894

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/ApplicationSettingsApi.java [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
-import com.fasterxml.jackson.databind.JsonNode;
-
 /**
  * This class implements the client side API for the GitLab Application Settings API.
  * See <a href="https://docs.gitlab.com/ee/api/settings.html">Application Settings API at GitLab</a> for more information.
  */
 public class ApplicationSettingsApi extends AbstractApi {
 
-    public ApplicationSettingsApi(GitLabApi gitLabApi) {
+    public ApplicationSettingsApi(final GitLabApi gitLabApi) {
```
Comment (jmini) [inline src/main/java/org/gitlab4j/api/ApplicationSettingsApi.java:17]: Why did you make all the method argument explicitly final?

I think the compiler was already treating those as implicitly final

File: pom.xml [hunk 2]
```diff
@@ -297,7 +296,8 @@
 				<configuration>
 					<signature>
 						<groupId>org.codehaus.mojo.signature</groupId>
-						<artifactId>java1${java.level}</artifactId>
+						<artifactId>java18</artifactId>
```
Comment (jmini) [inline pom.xml:299]: Per https://github.com/mojohaus/animal-sniffer/issues/62, keeping animal-sniffer when building with Java 11 does not really make sense.

Comment (gedeffe) [reply pom.xml:299]: So, we would be able to remove this configuration ?

**Comment by gedeffe** [comment]
> This is similar to the work that was done in #841 not?
> 
> As I wrote on #926, for me a solution could be to start working on those topic on a separated branch and have a cascade between the `5.x` branch and the `6.x` branch…

If I understand correctly, it would be better to target branch 6.x for these evolutions ?

File: pom.xml (hunk 2 above)
Comment (jmini) [reply pom.xml:299]: This is what I did in https://github.com/gitlab4j/gitlab4j-api/pull/946 (keeping you are co-author of the change)

**Comment by gedeffe** [comment]
@jmini , Thank you for integration in branch 6.x, especially as I made noise with automatic format of code (like adding final keyword).

Do I have to keep this PR alive, or should I edit it to target only 6.x branch or close it ?

**Comment by jmini** [comment]
@ajivemu: Please check if the release `6.0.0-rc.1` works for you and give feedback, so that we know if we are on the good path for a `6.0.0` release.

This version is using:

* minimal version: **Java 11** 
* Jakarta EE components with the `jakarta.*` packages instead of `javax.*`

**Comment by jmini** [comment]
@gedeffe https://github.com/gitlab4j/gitlab4j-api/pull/943#issuecomment-1504727995 :

> Do I have to keep this PR alive, or should I edit it to target only 6.x branch or close it ?

I have open 3 PRs with all your ideas from this PR.

You can review the last open one https://github.com/gitlab4j/gitlab4j-api/pull/950 and tell if something is missing.

For me this PR can be closed. Thank you very much for your work.

**Comment by gedeffe** [comment]
And I forgot to mention also here that version 6.0.0-rc.1 works for our use cases.
Thank you guys !

**Pull request closed without merging**